packages/FamilyTree/Reflection/FamilyRelationsAnalyzer.php

```diff
@@ -36,4 +64,97 @@ public function getChildrenOfClassReflection(ClassReflection $desiredClassReflec
 
         return $childrenClassReflections;
     }
+
+    /**
+     * @param Name|NullableType|PhpParserUnionType|null $propertyTypeNode
+     */
+    public function getPossibleUnionPropertyType(
+        Property $property,
+        Type $varType,
+        ?Scope $scope,
+        ?Node $propertyTypeNode
+    ): PropertyType {
+        if ($varType instanceof UnionType) {
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        if (! $scope instanceof Scope) {
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        /** @var ClassReflection $classReflection */
+        $classReflection = $scope->getClassReflection();
+        $ancestors = $classReflection->getAncestors();
+        $propertyName = $this->nodeNameResolver->getName($property);
+
+        $kindPropertyFetch = $property->isStatic()
+            ? StaticPropertyFetch::class
+            : PropertyFetch::class;
+
+        $className = $property->getAttribute(AttributeKey::CLASS_NAME);
+        foreach ($ancestors as $ancestor) {
+            $ancestorName = $ancestor->getName();
+            if ($ancestorName === $className) {
+                continue;
+            }
+
+            $fileName = (string) $ancestor->getFileName();
```

Comment:
This actually passed some problems in the past. The empty string in readFile() would cause an error.

The `false` return type should be handled instead of retyped

Reply:
fixed https://github.com/rectorphp/rector-src/pull/295/commits/01bde8591fa9fff8a7df49b6c14a281159dc7b8c